File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizSlider.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun BizSlider(
    value: Float,
    onValueChange: (Float) -> Unit,
    modifier: Modifier = Modifier,
    valueRange: ClosedFloatingPointRange<Float> = 0f..100f,
    steps: Int = 0,
    label: String? = null,
    showValue: Boolean = false,
    formatValue: ((Float) -> String)? = null,
    color: Color = Color(0xFF003399)
) {
    Column(
        modifier = modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        if (label != null || showValue) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                if (label != null) {
                    Text(
                        text = label,
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Medium,
                        color = Color(0xFF1A2A5E)
                    )
                } else {
                    Spacer(modifier = Modifier.width(1.dp))
                }

                if (showValue) {
                    val displayValue = formatValue?.invoke(value) ?: value.toInt().toString()
                    Text(
                        text = displayValue,
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Bold,
                        color = color
                    )
                }
            }
        }

        Slider(
            value = value,
            onValueChange = onValueChange,
            valueRange = valueRange,
            steps = steps,
            colors = SliderDefaults.colors(
                thumbColor = color,
                activeTrackColor = color,
                inactiveTrackColor = color.copy(alpha = 0.2f)
            ),
            modifier = Modifier.padding(horizontal = 8.dp)
        )
    }
}

File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizSlider.swift
import SwiftUI

public struct BizSlider: View {
    @Binding public var value: Double
    public var bounds: ClosedRange<Double>
    public var step: Double?
    public var label: String?
    public var showValue: Bool
    public var formatValue: ((Double) -> String)?
    public var color: Color
    
    public init(
        value: Binding<Double>,
        bounds: ClosedRange<Double> = 0...100,
        step: Double? = nil,
        label: String? = nil,
        showValue: Bool = false,
        formatValue: ((Double) -> String)? = nil,
        color: Color = Color(hex: "#003399")
    ) {
        self._value = value
        self.bounds = bounds
        self.step = step
        self.label = label
        self.showValue = showValue
        self.formatValue = formatValue
        self.color = color
    }
    
    public var body: some View {
        VStack(spacing: 4) {
            if label != nil || showValue {
                HStack {
                    if let label = label {
                        Text(label)
                            .font(.system(size: 14, weight: .medium))
                            .foregroundColor(Color(hex: "#1A2A5E"))
                    }
                    Spacer()
                    if showValue {
                        let displayValue = formatValue?(value) ?? String(format: "%.0f", value)
                        Text(displayValue)
                            .font(.system(size: 14, weight: .bold))
                            .foregroundColor(color)
                    }
                }
                .padding(.horizontal, 4)
            }
            
            if let step = step {
                Slider(value: $value, in: bounds, step: step)
                    .accentColor(color)
            } else {
                Slider(value: $value, in: bounds)
                    .accentColor(color)
            }
        }
    }
}
